### PR TITLE
Write okhttp collector client.

### PR DESCRIPTION
**Pull request by MatthewDolan**

### CODE REVIEW

File: okhttp/src/main/java/com/lightstep/tracer/shared/HttpCollectorClient.java [hunk 1]
```diff
@@ -2,18 +2,99 @@
...
-  void reconnect() {}
+    private static void shutdown(OkHttpClient client) {
+        client.dispatcher().executorService().shutdown();
+    }
 
-  @Override
-  void shutdown() {}
+    private static OkHttpClient start(long deadlineMillis) {
+        return new OkHttpClient.Builder()
+                .connectTimeout(deadlineMillis, TimeUnit.MILLISECONDS)
```
Comment (MatthewDolan) [inline okhttp/src/main/java/com/lightstep/tracer/shared/HttpCollectorClient.java]: I'm not 100% sure this is the right timeout we want.

Comment (frenchfrywpepper) [reply okhttp/src/main/java/com/lightstep/tracer/shared/HttpCollectorClient.java]: seems right

File: okhttp/src/main/java/com/lightstep/tracer/shared/HttpCollectorClient.java [hunk 2]
```diff
@@ -2,18 +2,99 @@
...
+        try {
+            return fromResponse(this.client().newCall(toRequest(request)).execute());
+        } catch (Exception e) {
+            tracer.error("Exception sending report to collector: ", e);
+            return null;
+        }
+    }
+
+    @Override
+    void reconnect() {
```
Comment (danielamiao) [inline okhttp/src/main/java/com/lightstep/tracer/shared/HttpCollectorClient.java]: Not blocking, more of a note here: reconnect is hacky-ish way we came up with to do client-side load balancing for GRPC connections (resetting connections every 5 minutes theoretically round robins through all the backend servers). This is not needed for HTTP clients and seems wasteful, so we will need to come up with a better solution, maybe use reflection to figure out client impl and skip the reconnect [here](https://github.com/lightstep/lightstep-tracer-java-common/blob/master/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java#L219)

Comment (MatthewDolan) [reply okhttp/src/main/java/com/lightstep/tracer/shared/HttpCollectorClient.java]: Ya, we should probably make sure we round robin the connections anyway. HTTP2 will have persistent connections so we will have the same problem we had with GRPC. Although there might be a setting in okhttp to tell it how long you allow connections to live.

**Pull request merged**